src/main.rs

```diff
@@ -201,11 +206,11 @@ fn main() {
     let file = File::open(filename).unwrap();
 
     // vectors to hold gene names and gene descriptions
-    let mut genes = Vec::<String>::new();
-    let mut descs = Vec::<String>::new();
+    let mut genes = vec![];
+    let mut descs = vec![];
 
     // HashMap holding counts of each feature type, eg, "("CDS", 5)"
-    let mut feature_map: HashMap<String, usize> = HashMap::new();
+    let mut feature_map = HashMap::new();
```

Comment:
These are all examples of places where Rust can figure out the type of the variable, and so we don't need to be explicit when declaring. It's more of a personal choice, there is nothing wrong in specifying them too, but coming from Python sometimes types can make code harder to read, so I tend to avoid being too explicit if it is not needed =]